### ISSUE TITLE
Write_SAV generates unreadable file when value code length is more than storage_width

### ISSUE THREAD

**Issue by nishant154** [Roche/pyreadstat#164]
Hi Team,
First of all thank you so much for your work around Pyreadstat... It has definitely made or lives a lot easier when working with SPSS generated files.

I wanted to highlight one issue that i have noticed in the **write_sav** function with value codes for String variables. If this issue has already been discussed or highlighted then please do point me the right direction.

I have shared a below sample code which generated an SAV file with a single string column.
the issue is, when i run this code it generates an sav file which cannot be read by readstat and cannot be read by spss as well.
it gives the error 

> Invalid file or has unsupported features.

`maindf=pd.DataFrame({'userid':['at_tempor','sed_vulputate',None,'blandit_aliquam'],'attempt':[1,2,3,4]})`

```
pyreadstat.write_sav(maindf,"./output_spss_file.zsav"
                     ,compress=True
                     ,variable_value_labels={'userid':{'at_tempor':'name1','sed_vulputate':'name2'
                                                       ,'blandit_aliquam':'name3','commodo_ullamcorper':'name4'}}
                     ,variable_display_width={'userid':20}
                     ,variable_format={'userid':'A20'}
                     ,variable_measure={'userid':'nominal'}
                    )
```
`df,meta=pyreadstat.read_sav("./output_spss_file.zsav")`

what i have been able to identify is that the length of the values in the dataframe column is less than the length of the last value code **commodo_ullamcorper** and the storage width seems to be the issue. 
If someone has an alternative to make this work then please do let me know.

**Setup Information:**
How did you install pyreadstat? PIP
Platform: Linux
Python Version : 3.6
Virtual ENV

**Comment by kaiyanchen** [comment]
hello @nishant154 , recently I met another issue when helping my team with the sav file.  it was a different issue: the label column changed after using write_sav.  
I solved it by downgrading the pyreadstat==1.1.4 to pyreadstat==1.1.2.  actually, I don't know what happened in the new version, but it did solve my issue. hope this will help you.

**Comment by nishant154** [comment]
Hi @kaiyanchen Thankyou for the suggestion,  i tried my example on both 1.1.2 and 1.1.3 and got the same error.

**Comment by ofajardo** [comment]
very intriguing, it seems that the trigger is to have a character value with more than a certain number of characters. The number of characters necessary to trigger the issue depends on the position of the value, if it is the first one for example you need more characters than if it is the fourth, in your example it is the fourth having more than 12 characters what causes the issue.

**Comment by nishant154** [comment]
Thanks a lot for looking into this @ofajardo

**Comment by ofajardo** [comment]
OK actually the error was on my side, I have to calculate what is the largest string in the variable to enter the storage size of the variable, I was not taking into consideration that the value label key can actually have an even larger length. Now corrected on dev, will be fixed with the next release.

**Comment by ofajardo** [comment]
solved in version 1.1.5
